ldpd/lde.c

```diff
@@ -1708,6 +1721,38 @@ on_get_label_chunk_response(uint32_t start, uint32_t end)
 		current_label_chunk = listtail(label_chunk_list);
 }
 
+void
+lde_free_label(uint32_t label)
+{
+	struct listnode	*node;
+	struct label_chunk	*label_chunk;
+	uint64_t		pos;
+
+	node = listhead(label_chunk_list);
+
+	while (node) {
+		label_chunk = listgetdata(node);
+
+		if (label_chunk != NULL)
+			if ((label <= label_chunk->end) && (label >= label_chunk->start)) {
```

Comment:
This should be `label < label_chunk->end` as the `end` label is not part of the range.

Also, please remove the redundant parentheses.

Reply:
@rwestphal ,  I  addressed all comments except this. I can see label chunks start and end values are like  16-79, 80-143,  144-207, etc.  So,  label_chunk->end is part of the range.